### ISSUE TITLE
new numpy version breaks childrenBounds in ViewBox

### ISSUE THREAD

**Issue by fedebarabas** [pyqtgraph/pyqtgraph#692]
numpy version '1.14.3'
pyqtgraph version '0.10.0'

When either `xr` or `yr` have boolean values, this error shows up:

```
bounds = QtCore.QRectF(xr[0], yr[0], xr[1]-xr[0], yr[1]-yr[0])
TypeError: numpy boolean subtract, the `-` operator, is deprecated, use the bitwise_xor, the `^` operator, or the logical_xor function instead.
```

I tried fixing it but failed because I could not figure out what a boolean value in those variables meant.

**Comment by pepasjc** [comment]
my solution is not the best but it works

`bounds = QtCore.QRectF(float(xr[0]), float(yr[0]), float(xr[1])-float(xr[0]), float(yr[1])-float(yr[0]))`

**Comment by j9ac9k** [comment]
Fixed with https://github.com/pyqtgraph/pyqtgraph/pull/876

**Comment by UsYer** [comment]
Hi,
this issue is not fixed with #876 and I'd like this issue to be reopened.
When plotting an array containing python bools or numpy bool_ a TypeError is raised:

>     p1.plot([bool(var) for var in range(0, 20)], pen=(255,0,0), name="Red curve")
  File "C:\develop\gitRepos\livetrace\third_party\pyqtgraph\pyqtgraph\graphicsItems\PlotItem\PlotItem.py", line 608, in plot
    self.addItem(item, params=params)
  File "C:\develop\gitRepos\livetrace\third_party\pyqtgraph\pyqtgraph\graphicsItems\PlotItem\PlotItem.py", line 485, in addItem
    self.vb.addItem(item, *args, **vbargs)
  File "C:\develop\gitRepos\livetrace\third_party\pyqtgraph\pyqtgraph\graphicsItems\ViewBox\ViewBox.py", line 410, in addItem
    self.updateAutoRange()
  File "C:\develop\gitRepos\livetrace\third_party\pyqtgraph\pyqtgraph\graphicsItems\ViewBox\ViewBox.py", line 862, in updateAutoRange
    childRange = self.childrenBounds(frac=fractionVisible)
  File "C:\develop\gitRepos\livetrace\third_party\pyqtgraph\pyqtgraph\graphicsItems\ViewBox\ViewBox.py", line 1355, in childrenBounds
    bounds = QtCore.QRectF(xr[0], yr[0], xr[1]-xr[0], yr[1]-yr[0])
TypeError: numpy boolean subtract, the `-` operator, is deprecated, use the bitwise_xor, the `^` operator, or the logical_xor function instead.
See the attached file for a minimal reproducing example.
I tried the current development branch.
[PlottingBoolBug.txt](https://github.com/pyqtgraph/pyqtgraph/files/3818841/PlottingBoolBug.txt)

**Comment by j9ac9k** [comment]
Just as I re-opened this issue, I saw that you created a new one, so I'll close this out again.  Discussion should take place in #1070
